File: 900/1883B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n, k;
        cin >> n >> k;

        string s;
        cin >> s;

        int one = 0;
        int odd = 0;

        unordered_map<char, int> mp;
        for (int i = 0; i < (int)s.size(); i++)
            mp[s[i]]++;

        int deleteCount = 0;
        for (int i = 0; i < (int)s.size(); i++) 
        {
            if (mp[s[i]] == 1) 
            {
                if (odd > 0) 
                {
                    if (one > 0)
                        deleteCount += 2;
                    else
                        deleteCount += 1;
                    one = 0;
                } 
                else 
                {
                    if (one > 0)
                        deleteCount += 1;
                    else
                        one = 1;
                }
            } 
            else if (mp[s[i]] % 2 == 1) 
            {
                if (one == 1) 
                {
                    one = 0;
                    deleteCount += 1;
                    odd = mp[s[i]];
                } 
                else 
                {
                    if (odd > 0) 
                    {
                        deleteCount += min(mp[s[i]], odd);
                        odd = max(mp[s[i]], odd);
                    } 
                    else
                        odd = mp[s[i]];
                }
            }
            // else even count, do nothing
            mp[s[i]] = 0;
        }

        if (deleteCount <= k)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}

File: 900/1883B/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int k = sc.nextInt();
            
            String s = sc.next();
            
            int one = 0;
            int odd = 0;
            
            HashMap<Character,Integer> mp = new HashMap<>();
            for(int i=0;i<s.length();i++)
            {
                if(mp.get(s.charAt(i)) == null)
                    mp.put(s.charAt(i), 1);
                else
                    mp.put(s.charAt(i), mp.get(s.charAt(i))+1);
            }
            
            int delete = 0;
            for(int i=0;i<s.length();i++)
            {
                if(mp.get(s.charAt(i)) == 1)
                {
                    if(odd > 0)
                    {
                        if(one > 0)
                            delete += 2;
                        else
                            delete += 1;
                        one = 0;
                    }
                    else
                    {
                        if(one > 0)
                            delete += 1;
                        else
                            one = 1;
                    }
                }
                else if(mp.get(s.charAt(i))%2 == 1)
                {
                    if(one == 1)
                    {
                        one = 0;
                        delete += 1;
                        odd = mp.get(s.charAt(i));
                    }
                    else
                    {
                        if(odd > 0)
                        {
                            delete += Math.min(mp.get(s.charAt(i)), odd);
                            odd = Math.max(mp.get(s.charAt(i)), odd);
                        }
                        else
                            odd = mp.get(s.charAt(i));
                    }
                }
                else
                {}
                mp.put(s.charAt(i), 0);
            }
            
            if(delete <= k)
                System.out.println("YES");
            else
                System.out.println("NO");
        }
    }
}